pallets/did/src/lib.rs

```diff
@@ -900,4 +981,23 @@ impl<T: Config> Pallet<T> {
 			.verify_signature(payload, signature)
 			.map_err(DidError::SignatureError)
 	}
+
+	/// Deletes DID details from storage, adds the identifier to the blacklisted
+	/// DIDs and frees the deposit.
+	fn delete_did(did_subject: DidIdentifierOf<T>) -> DispatchResult {
+		// `take` calls `kill` internally
+		let did_entry = Did::<T>::take(&did_subject).ok_or(Error::<T>::DidNotPresent)?;
+		// *** No Fail beyond this point ***
+
+		kilt_support::free_deposit::<AccountIdentifierOf<T>, CurrencyOf<T>>(&did_entry.deposit);
```

Comment:
Did we say we are not releasing the whole deposit OR making this extrinsic more expensive?

Reply:
The deposit is a reserved balance. It never leaves the account. So you probably mean that we rather withdraw the fee when we free the deposit instead of withdrawing it immediately?